trinity/main.py

```diff
@@ -0,0 +1,249 @@
+import argparse
+import asyncio
+import atexit
+from multiprocessing.managers import (
+    BaseManager,
+)
+import sys
+
+from evm.db.backends.level import LevelDB
+from evm.db.chain import ChainDB
+
+from evm.p2p.peer import (
+    LESPeer,
+    PeerPool,
+)
+
+from trinity.__version__ import __version__
+from trinity.chains import (
+    get_chain_protocol_class,
+    initialize_data_dir,
+    initialize_database,
+    is_data_dir_initialized,
+    is_database_initialized,
+    serve_chaindb,
+)
+from trinity.cli import console
+from trinity.constants import (
+    ROPSTEN,
+    SYNC_LIGHT,
+)
+from trinity.db.chain import ChainDBProxy
+from trinity.db.base import DBProxy
+from trinity.utils.chains import (
+    ChainConfig,
+)
+from trinity.utils.ipc import (
+    wait_for_ipc,
+    kill_process_gracefully,
+)
+from trinity.utils.logging import (
+    setup_trinity_logging,
+    with_queued_logging,
+)
+from trinity.utils.mp import (
+    ctx,
+)
+
+
+DEFAULT_LOG_LEVEL = 'info'
+LOG_LEVEL_CHOICES = (
+    'debug',
+    'info',
+)
+
+
+parser = argparse.ArgumentParser(description='Trinity')
+
+# enable `trinity --version`
+parser.add_argument('--version', action='version', version=__version__)
+
+# set global logging level
+parser.add_argument(
+    '-l',
+    '--log-level',
+    choices=LOG_LEVEL_CHOICES,
+    default=DEFAULT_LOG_LEVEL,
+    help="Sets the logging level",
+)
+
+# options for running chains
+parser.add_argument(
+    '--ropsten',
+    action='store_true',
+    help="Ropsten network: pre configured proof-of-work test network",
+)
+parser.add_argument(
+    '--light',  # TODO: consider --sync-mode like geth.
+    action='store_true',
+)
+parser.add_argument(
+    '--trinity-root-dir',
+    help=(
+        "The filesystem path to the base directory that trinity will store it's "
+        "information.  Default: $XDG_DATA_HOME/.local/share/trinity"
+    ),
+)
+parser.add_argument(
+    '--data-dir',
+    help=(
+        "The directory where chain data is stored"
+    ),
+)
+parser.add_argument(
+    '--nodekey',
+    help=(
+        "Hexadecimal encoded private key to use for the nodekey"
+    )
+)
+parser.add_argument(
+    '--nodekey-path',
+    help=(
+        "The filesystem path to the file which contains the nodekey"
+    )
+)
+
+# Add console sub-command to trinity CLI.
+subparser = parser.add_subparsers(dest='subcommand')
+console_parser = subparser.add_parser('console', help='start the trinity REPL')
+console_parser.add_argument(
+    '--vanilla-shell',
+    action='store_true',
+    default=False,
+    help='start a native Python shell'
+)
+console_parser.set_defaults(func=console)
```

Comment:
IIUC, this will run the `LightChain` in a thread and drop into a python shell, right? In that case, should we remove evm/lightchain_shell.py which was just  a quick hack anyway?

Reply:
Yes, we can remove `evm/lightchain_shell.py`.
 
Also, anything console related can largely be ignored in this PR as it will be significantly overhauled very soon to instead use IPC connections to access the database.